heiseonline/shariff
# Add new service Buffer

This pull request (P) adds the new service "Buffer" for [buffer](https://buffer.com/) to Shariff.

Buffer is something very similar to AddThis.

The URL for the button I found here: [https://github.com/bradvin/social-share-urls#buffer](https://github.com/bradvin/social-share-urls#buffer).

Buffer also provides share counts. A PR to add the "Buffer" service to Shariff-Backend-PHP can be found here: [https://github.com/heiseonline/shariff-backend-php/pull/155](https://github.com/heiseonline/shariff-backend-php/pull/155).

## Dependencies

This PR requires PR #360 in order to have an icon for Buffer.

## About button colors

The original sharing button of Buffer has black logo on grey background, and their sharing count has some kind of green background, see here: [https://buffer.com/extras/button](https://buffer.com/extras/button):
![buffer-button-original](https://user-images.githubusercontent.com/7413183/54879735-d16b8680-4e3c-11e9-957a-5ebf88cafc1d.png)
This design does not really fit to Shariff.

On their website, buffer.com have sharing buttons on their own pages.
They look as follows:
![sharing-buttons-at-buffer-com-pages](https://user-images.githubusercontent.com/7413183/55288453-56fcb280-53b8-11e9-9257-b2d9e0dfde18.png)
There is no hover effect.

Another thing I found was a Wordpress plugin with buttons looking as follows:
![screenshot-1](https://user-images.githubusercontent.com/7413183/55288466-7dbae900-53b8-11e9-9acc-df08f79c8a94.png)

(Buffer button is the leftmost one in both examples).

So I've decided for following button colors, with color numbers picked from the official buffer icon:

Standard theme:
![buffer-button-new](https://user-images.githubusercontent.com/7413183/55289084-e9ee1a80-53c1-11e9-8455-d544df0e624f.png)

Grey theme:
![buffer-button-new-grey](https://user-images.githubusercontent.com/7413183/55289093-fa9e9080-53c1-11e9-8a21-e927252b1716.png)

Standard or grey theme hovered:
![buffer-button-new-hover](https://user-images.githubusercontent.com/7413183/55289088-f3778280-53c1-11e9-9acd-ce0fecfedcb8.png)

White theme:
![buffer-button-new-white](https://user-images.githubusercontent.com/7413183/55289104-186bf580-53c2-11e9-9146-cec28728151e.png)

White theme hovered:
![buffer-button-new-white-hover](https://user-images.githubusercontent.com/7413183/55289259-6c77d980-53c4-11e9-96de-5d52b8281c40.png)

## How to test:

The test is done with a branch which includes the changes from PR #360 and this PR here.

Run the shariff demo site locally on a Linux host which has git and npm installed with the branch of this PR as follows:

```sh
$ mkdir shariff-test-pr361
$ cd shariff-test-pr361
$ git clone https://github.com/richard67/shariff-plus.git
$ cd shariff-plus
$ git checkout upstream-test-buffer
$ git pull origin upstream-test-buffer
$ npm install
$ npm run dev
```
Check the result in the browser, URL=`http://localhost:3000/`.

#### Check the result

Verify that a new button for Buffer has been added at the end of examples 2, 3 and 4 of the demo, that the buffer icon fits to the brand icon on [https://buffer.com/](https://buffer.com/), that the button colors are as described above, and that the sharing links work and use the right title and URL.

#### Clean up after the test
```sh
$ cd ../..
$ rm -rf ./shariff-test-pr361
```
